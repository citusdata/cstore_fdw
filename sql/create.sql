--
-- Test the CREATE statements related to cstore_fdw.
--


-- Install cstore_fdw
CREATE EXTENSION cstore_fdw;

CREATE SERVER cstore_server FOREIGN DATA WRAPPER cstore_fdw;


-- Validator tests
CREATE FOREIGN TABLE test_validator_invalid_option () 
	SERVER cstore_server 
	OPTIONS(bad_option_name '1'); -- ERROR

CREATE FOREIGN TABLE test_validator_invalid_stripe_row_count () 
	SERVER cstore_server
	OPTIONS(stripe_row_count '0'); -- ERROR

CREATE FOREIGN TABLE test_validator_invalid_block_row_count () 
	SERVER cstore_server
	OPTIONS(block_row_count '0'); -- ERROR

CREATE FOREIGN TABLE test_validator_invalid_compression_type () 
	SERVER cstore_server
	OPTIONS(filename 'data.cstore', compression 'invalid_compression'); -- ERROR

-- filename option is not supported anymore
CREATE FOREIGN TABLE test_invalid_file_path ()
	SERVER cstore_server
	OPTIONS(filename 'data.cstore'); --ERROR

-- valid logging options are true, and false
CREATE FOREIGN TABLE test_invalid_log_option ()
	SERVER cstore_server
	OPTIONS(logging 'enabled'); --ERROR

-- Create uncompressed table
CREATE FOREIGN TABLE contestant (handle TEXT, birthdate DATE, rating INT,
	percentile FLOAT, country CHAR(3), achievements TEXT[])
	SERVER cstore_server;

-- Create compressed table with automatically determined file path
CREATE FOREIGN TABLE contestant_compressed (handle TEXT, birthdate DATE, rating INT,
	percentile FLOAT, country CHAR(3), achievements TEXT[])
	SERVER cstore_server
	OPTIONS(compression 'pglz');

-- Test that querying an empty table works
ANALYZE contestant;
SELECT count(*) FROM contestant;
