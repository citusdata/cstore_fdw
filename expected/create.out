--
-- Test the CREATE statements related to cstore_fdw.
--
-- Install cstore_fdw
CREATE EXTENSION cstore_fdw;
CREATE SERVER cstore_server FOREIGN DATA WRAPPER cstore_fdw;
-- Validator tests
CREATE FOREIGN TABLE test_validator_invalid_option () 
	SERVER cstore_server 
	OPTIONS(bad_option_name '1'); -- ERROR
ERROR:  invalid option "bad_option_name"
HINT:  Valid options in this context are: compression, stripe_row_count, block_row_count, logging
CREATE FOREIGN TABLE test_validator_invalid_stripe_row_count () 
	SERVER cstore_server
	OPTIONS(stripe_row_count '0'); -- ERROR
ERROR:  invalid stripe row count
HINT:  Stripe row count must be an integer between 1000 and 10000000
CREATE FOREIGN TABLE test_validator_invalid_block_row_count () 
	SERVER cstore_server
	OPTIONS(block_row_count '0'); -- ERROR
ERROR:  invalid block row count
HINT:  Block row count must be an integer between 1000 and 100000
CREATE FOREIGN TABLE test_validator_invalid_compression_type () 
	SERVER cstore_server
	OPTIONS(filename 'data.cstore', compression 'invalid_compression'); -- ERROR
ERROR:  invalid option "filename"
HINT:  Valid options in this context are: compression, stripe_row_count, block_row_count, logging
-- filename option is not supported anymore
CREATE FOREIGN TABLE test_invalid_file_path ()
	SERVER cstore_server
	OPTIONS(filename 'data.cstore'); --ERROR
ERROR:  invalid option "filename"
HINT:  Valid options in this context are: compression, stripe_row_count, block_row_count, logging
-- valid logging options are true, and false
CREATE FOREIGN TABLE test_invalid_log_option ()
	SERVER cstore_server
	OPTIONS(logging 'enabled'); --ERROR
ERROR:  invalid logging flag
HINT:  Valid options are: true, false
-- Create uncompressed table
CREATE FOREIGN TABLE contestant (handle TEXT, birthdate DATE, rating INT,
	percentile FLOAT, country CHAR(3), achievements TEXT[])
	SERVER cstore_server;
-- Create compressed table with automatically determined file path
CREATE FOREIGN TABLE contestant_compressed (handle TEXT, birthdate DATE, rating INT,
	percentile FLOAT, country CHAR(3), achievements TEXT[])
	SERVER cstore_server
	OPTIONS(compression 'pglz');
-- Test that querying an empty table works
ANALYZE contestant;
SELECT count(*) FROM contestant;
 count 
-------
     0
(1 row)

